geo/src/algorithm/contains/geometry.rs

```diff
@@ -7,19 +7,19 @@ use crate::*;
 
 impl<T> Contains<Coordinate<T>> for Geometry<T>
 where
-    T: GeoNum,
+    T: GeoFloat,
```

Comment:
Note this is a breaking change - I'm dropping support for `Contains` for integer Polygons, and thus integer Geometries.

If this is controversial, I could revert this part, but IMO it's more important to fix #513 than to support integer geometries.

Reply:
@rmanoka  - You've previously convinced me that integer geometries are neat, so I'd like to support them, but it seems unreasonable to leave things broken for the mainstream use case in order to hold on to a possibility of something more esoteric.

Assuming [specialization](https://github.com/rust-lang/rust/issues/31844) won't be finalized any time soon, do you have any thoughts on how we could support both? 

I'd be curious to see how well something like this geometry graph approach lends itself to integer geometries. I took a brief spelunking just to see what would break from a code POV: https://github.com/georust/geo/commit/cf56814202c3aefd297ad1febaef541fe1c20640

The large commented out portions are the things that don't compile as GeoNum.

I don't have any experience working with integer geometries - some of it seems readily addressable, but the biggest immediate hangup for me is that the approach in this PR relies heavily on knowing where lines intersect - is their a corollary in integer geometries? I'm assuming you can't just round.

Reply:
I should say, it is my preference to get this mega PR merged without adding even more changes to support Relate for integers. I think the Relate functionality is useful even if it's just for Floats (which I expect is almost all users).

As I said, if it's controversial to drop support for Contains<Integer>, I could revert this portion. I'd still add the new relate trait, just for Floats, but revert the fix for the Contains edge cases. I'd be OK with that approach if that's your preference @rmanoka.

Reply:
@michaelkirk Relate is a brilliant trait and I think we should definitely add it even if can only supports `GeoFloat`.  What I was wondering is whether we could leave the previous implementations of `Contains` as long as they pass all test cases as `GeoNum` subsumes `GeoFloat`.

For the above case, is it possible to use our hand-rolled `Contains<Coordinate>` and `Contains<Point>` implementations for the geometries?  From what I recall, we did handle the edge cases, and hopefully the JTS test suite will point out any edge cases we missed out for these.  I do agree `Contains<LineString>`, and `Contains<Polygon>` should use the geomgraph techniques you've implemented as there were corner cases we couldn't handle easily earlier.

Reply:
And just to be clear, I'm still removing the API for other GeoNum backed variants, like `LineString<GeoNum>`, which now only support `LineString<GeoFloat>`